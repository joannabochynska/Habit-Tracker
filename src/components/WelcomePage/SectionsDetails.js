export const entrySectionHeader = "Build Habits That Stick";

export const entrySectionText =
  `Habit Tracker helps you learn the most effective process,
  increase self-awareness and set priorities
  so achieving your goals is easier than ever.`;

export const processSectionHeader = "Use The Proven Process";

export const processSectionText =
    `Habit Loop from "The Power of Habit" bu Charles Duhigg is a process
    of forming a habit. First, there is a cue, a trigger that tells your brain
    to go into automatic mode and which habit to use. Then there is a routine
    which can be physical or mental or emotional. Finally, there is a reward
    which helps your brain figure out if this particular loop is worth
    remembering for the future.`;

export const writingSectionHeader = "Power of Writing Down";

export const writingSectionText =
    `Journaling daily is a powerful tool which helps you evaluate your experiences
    and increase self-awareness. Habit Tracker is based on the idea of Buller Journal
    by Ridel Carroll. It is descripted as a mindfullness practice disguised
    as productivity system.`;

export const prioritySectionHeader = "Calm the Chaos by Setting Priorities";

export const prioritySectionText =
  `When we don't set priorities, we tend to follow the path of least resistance.
  Choose 3 most important habits and focus on them to get the results. We believe
  that saying "no" to some things determines what you can actually do.`;