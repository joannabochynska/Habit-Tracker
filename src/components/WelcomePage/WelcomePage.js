import React, { Component } from 'react';
import styles from './WelcomePage.module.scss';
import WelcomeSectionComponent from './WelcomeSection/WelcomeSection';
import entrySectionImage from '../../images/undraw_High_five.svg';
import processSectionImage from '../../images/habit_loop.png';
import writingSectionImage from '../../images/undraw_to_do_list.svg';
import prioritySectionImage from '../../images/undraw_country_side.svg';
import {
  entrySectionHeader,
  entrySectionText,
  processSectionHeader,
  processSectionText,
  writingSectionHeader,
  writingSectionText,
  prioritySectionHeader,
  prioritySectionText,
} from './SectionsDetails';

class WelcomePage extends Component {
  render() {
    return (
      <div>
        <div className={styles.grayBackground}>
          <WelcomeSectionComponent
            header={entrySectionHeader}
            text={entrySectionText}
            image={entrySectionImage}
            textPlacement="left"
          />
        </div>
        <div className={styles.violetBackground}>
          <WelcomeSectionComponent
            header={processSectionHeader}
            text={processSectionText}
            image={processSectionImage}
            textPlacement="right"
          />
        </div>
        <div className={styles.whiteBackground}>
          <WelcomeSectionComponent
            header={writingSectionHeader}
            text={writingSectionText}
            image={writingSectionImage}
            textPlacement="left"
          />
        </div>
        <div className={styles.grayBackground}>
          <WelcomeSectionComponent
            header={prioritySectionHeader}
            text={prioritySectionText}
            image={prioritySectionImage}
            textPlacement="right"
          />
        </div>
      </div>
    );
  }
}

export default WelcomePage;
