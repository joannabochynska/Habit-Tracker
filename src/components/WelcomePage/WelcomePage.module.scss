@import "../../scss/constants";

.grayBackground {
    background-color: $grayBackgroundColor;
}

.violetBackground {
    background-color: $violetBackgroundColor;
}

.whiteBackground {
    background-color: $whiteBackgroundColor;
}
