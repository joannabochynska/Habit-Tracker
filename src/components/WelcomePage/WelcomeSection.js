import React from 'react';
import PropTypes from 'prop-types';
import styles from './WelcomeSection.module.scss';

const renderImage = (image, header) => (
    <div className={styles.imageSection}>
        <img src={ image } alt={ header }/>
    </div>
);

const renderDetails = (header, text) => (
    <div className={styles.details}>
        <h4>{ header }</h4>
        <p>{ text }</p>
    </div>
);

const WelcomeSectionComponent = ({ header, text, image, textPlacement }) => (
    <div className={styles.section}>
        { textPlacement === "left"
            ? <div className={ `${styles.row} ${styles.left}`}>
                { renderDetails(header, text) }
                { renderImage(image, header) }
            </div>
            : <div className={ `${styles.row} ${styles.right}`}>
                { renderImage(image, header) }
                { renderDetails(header, text) }
            </div>
        }
    </div>
);

WelcomeSectionComponent.propTypes = {
    header: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    textPlacement: PropTypes.oneOf(['left', 'right']).isRequired,
}

export default WelcomeSectionComponent;