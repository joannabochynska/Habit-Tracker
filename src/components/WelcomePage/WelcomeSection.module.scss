@import "../../scss/constants";

.section {
    height: 650px;
    align-items: center;
    display: grid;
    grid-template-columns: 5% auto 5%;
}

.row {
    height: 500px;
    grid-column: 2;
    display: grid;
    grid-template-columns: 45% 10% 45%;
    align-items: center;

    .details {
        align-self: start;

        h4 {
            color: $headersTextColor;
            font-size: 56px;
            font-family: $mainFont;
            font-weight: 400;
            margin-bottom: 0;
        }

        p {
            margin: 30px 0;
            color: $textColor;
            font-family: $secondFont;
            font-size: 20px;
            line-height: 1.5;
        }
    }

    .imageSection {
        img {
            height: 400px;
            width: 100%;
        }
    }
}


.left {
    .details {
        grid-column: 1;
    }

    .imageSection {
        grid-column: 3;
    }
}

.right {
    .details {
        grid-column: 3;
    }

    .imageSection {
        grid-column: 1;
    }
}
