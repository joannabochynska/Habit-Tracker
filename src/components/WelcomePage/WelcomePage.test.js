import React from 'react';
import renderer from 'react-test-renderer';
import WelcomePage from './WelcomePage';

test('renders correctly', () => {
  const tree = renderer
    .create(<WelcomePage />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
