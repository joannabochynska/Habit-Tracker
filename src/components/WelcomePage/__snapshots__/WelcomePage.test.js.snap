// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="grayBackground"
  >
    <div
      className="section"
    >
      <div
        className="row left"
      >
        <div
          className="details"
        >
          <h4>
            Build Habits That Stick
          </h4>
          <p>
            Habit Tracker helps you learn the most effective process,
  increase self-awareness and set priorities
  so achieving your goals is easier than ever.
          </p>
        </div>
        <div
          className="imageSection"
        >
          <img
            alt="Build Habits That Stick"
            src="undraw_High_five.svg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="violetBackground"
  >
    <div
      className="section"
    >
      <div
        className="row right"
      >
        <div
          className="imageSection"
        >
          <img
            alt="Use The Proven Process"
            src="habit_loop.png"
          />
        </div>
        <div
          className="details"
        >
          <h4>
            Use The Proven Process
          </h4>
          <p>
            Habit Loop from "The Power of Habit" bu Charles Duhigg is a process
    of forming a habit. First, there is a cue, a trigger that tells your brain
    to go into automatic mode and which habit to use. Then there is a routine
    which can be physical or mental or emotional. Finally, there is a reward
    which helps your brain figure out if this particular loop is worth
    remembering for the future.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="whiteBackground"
  >
    <div
      className="section"
    >
      <div
        className="row left"
      >
        <div
          className="details"
        >
          <h4>
            Power of Writing Down
          </h4>
          <p>
            Journaling daily is a powerful tool which helps you evaluate your experiences
    and increase self-awareness. Habit Tracker is based on the idea of Buller Journal
    by Ridel Carroll. It is descripted as a mindfullness practice disguised
    as productivity system.
          </p>
        </div>
        <div
          className="imageSection"
        >
          <img
            alt="Power of Writing Down"
            src="undraw_to_do_list.svg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="grayBackground"
  >
    <div
      className="section"
    >
      <div
        className="row right"
      >
        <div
          className="imageSection"
        >
          <img
            alt="Calm the Chaos by Setting Priorities"
            src="undraw_country_side.svg"
          />
        </div>
        <div
          className="details"
        >
          <h4>
            Calm the Chaos by Setting Priorities
          </h4>
          <p>
            When we don't set priorities, we tend to follow the path of least resistance.
  Choose 3 most important habits and focus on them to get the results. We believe
  that saying "no" to some things determines what you can actually do.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
